src/SampleLibraryUI/Examples/ButtonCustomNodeModel.cs

```diff
@@ -0,0 +1,336 @@
+     * associated NodeViewModel which you should never need to
+     * edit. So here we will create a data binding between 
+     * properties on our class and our custom UI.
+    */
+
+    // The NodeName attribute is what will display on 
+    // top of the node in Dynamo
+    [NodeName("Button Custom Node Model")]
+    // The NodeCategory attribute determines how your
+    // node will be organized in the library. You can
+    // specify your own category or by default the class
+    // structure will be used.  You can no longer 
+    // add packages or custom nodes to one of the 
+    // built-in OOTB library categories.
+    [NodeCategory("SampleLibraryUI.Examples")]
+    // The description will display in the tooltip
+    // and in the help window for the node.
+    [NodeDescription("A sample UI node which displays custom UI.")]
+    // Specifying InPort and OutPort types simply
+    // adds these types to the help window for the
+    // node when hovering the name in the library.
+    [InPortTypes("string")]
+    [OutPortTypes("string")]
+    // Add the IsDesignScriptCompatible attribute to ensure
+    // that it gets loaded in Dynamo.
+    [IsDesignScriptCompatible]
+    public class ButtonCustomNodeModel : NodeModel
+    {
+        #region private members
+
+        private string buttonText;
+        private string windowText;
+        private const string defaultButtonText = "Default Button Text";
+        private const string defaultWindowText = "Default Window Text";
+
+        #endregion
+
+        #region properties
+
+        /// <summary>
+        /// Text that will appear on the button
+        /// on our node.
+        /// </summary>
+        public string ButtonText
+        {
+            get { return buttonText; }
+            set
+            {
+                buttonText = value;
+
+                // Raise a property changed notification
+                // to alert the UI that an element needs
+                // an update.
+                RaisePropertyChanged("ButtonText");
+            }
+        }
+
+        /// <summary>
+        /// Text that will appear in the message
+        /// box when button is pressed.
+        /// </summary>
+        public string WindowText
+        {
+            get { return windowText; }
+            set
+            {
+                windowText = value;
+            }
+        }
+
+        /// <summary>
+        /// DelegateCommand objects allow you to bind
+        /// UI interaction to methods on your data context.
+        /// </summary>
+        [JsonIgnore]
+        [IsVisibleInDynamoLibrary(false)]
+        public DelegateCommand ButtonCommand { get; set; }
+
+        #endregion
+
+        #region constructor
+
+        /// <summary>
+        /// The constructor for a NodeModel is used to create
+        /// the input and output ports and specify the argument
+        /// lacing.
+        /// </summary>
+        public ButtonCustomNodeModel()
```

Comment:
can you mention that this gets invoked when the node is added to the graph from the library or via copy paste?